### DISCUSSION TITLE
NModal 的 trap-focus 选项默认启用产生了意外的结果

### DISCUSSION THREAD

**Discussion by ouuan** [tusen-ai/naive-ui#2715] [Q&A] [answered]
我的实际场景是 `NModal` 内是一个 `NScrollbar`，里面开头有一个链接，剩下是纯文本，然后我发现每次切换窗口再回来，就会自动聚焦到开头那个链接导致滚动条位置改变，找了很久才发现是 `trap-focus` 选项导致的。

我不是很了解这个选项的作用，是为了防止聚焦到 `NModal` 的外面来保证其 modal 吗？就如我上面描述的情况，在失焦时聚焦到 `NModal` 内可聚焦的元素可能产生意外的结果，`trap-focus` 是否有必要默认启用，能否改成默认不启用，或者至少在文档里加以说明？

**Comment by 07akioni** [comment, marked as answer]
会在文档里进行说明，默认启用是对于 a11y 的考虑

**Reply by 07akioni** [reply]
为了 tab 切换聚焦元素能留在 modal 里面

**Reply by 07akioni** [reply]
可以把 `auto-focus` 设为 `false`

**Comment by ouuan** [comment]
> 为了 tab 切换聚焦元素能留在 modal 里面

这个一定要在失焦时聚焦到 NModal 内可聚焦元素才能实现吗 🤔 只监听 Tab keydown 会有什么问题吗（或者尝试过其它解决方案吗？）

**Reply by 07akioni** [reply]
还有鼠标点在外面要重新聚焦，具体你去看 https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html
